runtime/lua/tohtml.lua

```diff
@@ -0,0 +1,1263 @@
+---@defgroup vim.tohtml
+
+-- The HTML conversion script is different from Vim's one. If you want to use
+-- Vim's TOhtml converter, download it from the vim GitHub repo.
+-- Here are the Vim files related to this functionality:
+-- - https://github.com/vim/vim/blob/master/runtime/syntax/2html.vim
+-- - https://github.com/vim/vim/blob/master/runtime/autoload/tohtml.vim
+-- - https://github.com/vim/vim/blob/master/runtime/plugin/tohtml.vim
+--
+-- Main differences between this and the vim version:
+-- - No "ignore some visual thing" settings (just set the right Vim option)
+-- - No support for legacy web engines
+-- - No support for legacy encoding (supports only UTF-8)
+-- - No interactive webpage
+-- - No specifying the internal html (no XHTML, no use_css=false)
+-- - No multiwindow diffs
+-- - No ranges
+--
+-- Remarks:
+-- - Not all visuals are supported, so it may differ.
+
+--- @class vim.tohtml.opt
+--- @field title? string|false
+--- @field number_lines? boolean
+--- @field font? string[]|string
+--- @field width? number
+--- @class vim.tohtml.state.global
+--- @field background string
+--- @field foreground string
+--- @field title string|false
+--- @field font string
+--- @field highlights_name table<number,string>
+--- @field conf vim.tohtml.opt
+--- @class vim.tohtml.state:vim.tohtml.state.global
+--- @field style vim.tohtml.styletable
+--- @field tabstop string|false
+--- @field opt vim.wo
+--- @field winnr number
+--- @field bufnr number
+--- @field width number
+--- @field buflen number
+--- @class vim.tohtml.styletable
+--- @field [number] vim.tohtml.line (number: (1-index, exclusive))
+--- @class vim.tohtml.line
+--- @field virt_lines {[number]:{[1]:string,[2]:number}[]}
+--- @field pre_text string[][]
+--- @field hide? boolean
+--- @field [number] vim.tohtml.cell? (number: (1-index, exclusive))
+--- @class vim.tohtml.cell
+--- @field [1] number[] start
+--- @field [2] number[] close
+--- @field [3]  any[][] virt_text
+--- @field [4]  any[][] overlay_text
+
+local HIDE_ID = -1
+
+--- @type table<number,string>
+local cterm_color_cache = {}
+local background_color_cache = nil
+local foreground_color_cache = nil
```

Comment:
What are the type of these?

Reply:
`string?`
Added the type hints to code. (If it's that you're after)